packages/DataTable/src/hooks/useTableHeight.ts

```diff
@@ -0,0 +1,37 @@
+import React from "react";
+
+export default function useTableHeight(
+  tableRef: React.RefObject<HTMLDivElement>,
+  itemCount?: number
+): { getTableHeight: (maxHeight: number) => number } {
+  const tableHeightRef = React.useRef<number | undefined>();
+
+  React.useLayoutEffect(() => {
+    if (!tableRef.current) {
```

Comment:
Is it ok if we just return `maxHeight` when the `itemCount` is larger than `100` ? just for reduce the calculation

Reply:
Actually, the calculation should only occur once per `itemCount` change. 
What we could do, though, is to skip the calculation if the `tableHeightRef.current` is already set and itemCount is more than 100 (for instance)